spring-cloud-gcp-storage/src/main/java/org/springframework/cloud/gcp/storage/GoogleStorageResource.java

```diff
@@ -93,7 +110,18 @@ private Blob resolve() throws IOException {
 		URI uri = getURI();
 		BlobId blobId = BlobId.of(uri.getHost(),
 				uri.getPath().substring(1, uri.getPath().length()));
-		return this.storage.get(blobId);
+		Blob blob = this.storage.get(blobId);
+		if ((blob == null || !blob.exists()) && this.createBlobIfNotExists) {
+			return this.storage.create(BlobInfo.newBuilder(blobId).build());
```

Comment:
We should only create the blob on an attempt to write to it, not on just any `resovle()`

Reply:
+1 for this point. Right, no reason to create if no request for that.